src/vm/assembly.cpp

```diff
@@ -2588,6 +2588,7 @@ HRESULT RunMain(MethodDesc *pFD ,
             // Set the return value to 0 instead of returning random junk
             *pParam->piRetVal = 0;
             threadStart.Call(&stackVar);
+            *pParam->piRetVal = GetLatchedExitCode();
```

Comment:
@joktas It seems that there is no plan to support System.Environment.ExitCode property. If so, I'll close this PR.